File: app/src/main/java/com/intetm/comicbreeze/service/database/model/Comic.java
package com.intetm.comicbreeze.service.database.model;

import java.util.List;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;


@Entity
public class Comic {

    @JsonIgnore
    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    @Column(name = "name", nullable = false)
    public String name;

    @Column(name = "about", nullable = false)
    public String about;

    @Column(name = "url", nullable = false)
    public String url;

    @Column(name = "img_url", nullable = false)
    public String imgUrl;

    @Column(name = "page_count", nullable = false)
    public int pageCount;

    @JsonIgnore
    @Column(name = "expected_count", nullable = false)
    public int expectedCount;

    @JsonIgnore
    @OneToMany(mappedBy = "comic", fetch = FetchType.LAZY)
    public List<Page> pages;

    public Comic() {
    }

    public Comic(Long id) {
        this.id = id;
    }
}

File: app/src/main/java/com/intetm/comicbreeze/service/database/model/Comic.kt
package com.intetm.comicbreeze.service.database.model

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey


@Entity
class Comic {

    @PrimaryKey
    var id: Long? = null

    @ColumnInfo(name = "name")
    var name: String? = null

    @ColumnInfo(name = "about")
    var about: String? = null

    @ColumnInfo(name = "url")
    var url: String? = null

    @ColumnInfo(name = "img_url")
    var imgUrl: String? = null

    @ColumnInfo(name = "page_count")
    var pageCount: Int = 0
}
